### DISCUSSION TITLE
Relative imports

### DISCUSSION THREAD

**Discussion by aslilac** [gleam-lang/gleam#1490] [Ideas & suggestions]
(Bringing this here from Discord)

```gleam
// src/a.gleam
import a/b

// Now has stuff from src/a/b.gleam available
```

```gleam
// src/a/b.gleam
import ./c

// Now has stuff from src/a/c.gleam available
```

^ Could potentially be nice to have

**Comment by lpil** [comment]
Very nice, thank you. I could see us potentially adding this. Do you think one should be able to use `../../` etc?

**Reply by aslilac** [reply]
Yeah, those would be nice too!

**Reply by lpil** [reply]
After some thinking I'm not as sure about `../`. With `./` there's a clear relationship between the different modules, while with `../` the module is reaching outside its namespace.

What other languages have relative imports? Python and JS spring to mind, both support `../` and `./`

edit: I've made a poll here https://twitter.com/louispilfold/status/1491739342955552769

**Reply by aslilac** [reply]
Rust has this via `super::`. Different syntax, but same end result.

Plus you can already reach out of your namespace with the existing import system, you just have to know the absolute path. `../` wouldn't be a fundamental change in that regard, it would just make it easier.

**Reply by lpil** [reply]
Aye, that making it easier is the thing to consider here. With Gleam I'm attempting to create a "pit of success" where the most obvious or easiest approach is the best one. It's good to be conservative with features and consider how they'll influence people writing Gleam.

**Reply by inoas** [reply]
I am in favour of adding relative imports below the namespace path but not upwards of it.
Going up a path then down can make you end up anywhere and will make things hard to track.
